### PR TITLE
Thread locking

### PR DESCRIPTION
**Pull request by gandrewstone**
solves 2 locking / thread contention issues

### CODE REVIEW

**Review by ptschip — APPROVED** [review]
Already reviewed this yesterday which was in another PR...Looks good.

Didn't see the second commit..will continue to review

File: src/net.h [hunk 1]
```diff
@@ -568,7 +568,7 @@ class CNode
 
     // inventory based relay
     CRollingFastFilter<4 * 1024 * 1024> filterInventoryKnown;
-    std::vector<CInv> vInventoryToSend;
+    std::vector<CInv> vInventoryToSend GUARDED_BY(cs_inventory);
     CCriticalSection cs_inventory;
     int64_t nNextInvSend;
```
Comment (ptschip) [inline src/net.h:573]: Oh, didn't see this section.  I think you need to put the definition for cs_inventory above any guarded by statements...some compilers need that order

-std::vector<CInv> vInventoryToSend;
-CCriticalSection cs_inventory;
+CCriticalSection cs_inventory;
+std::vector<CInv> vInventoryToSend GUARDED_BY(cs_inventory);

Comment (gandrewstone) [reply src/net.h:573]: Ok will swap

**Review by ptschip — APPROVED** [review]
Nice catch on finding the missing lock.

**Pull request merged**